libvast/src/view.cpp

```diff
@@ -192,6 +192,49 @@ bool type_check(const type& t, const data_view& x) {
   return caf::holds_alternative<caf::none_t>(x) || caf::visit(f, t);
 }
 
+namespace {
+
+// Checks whether the left-hand side is contained in the right-hand side.
+struct contains_predicate {
+  template <class T, class U>
+  bool operator()(T lhs, U rhs) {
+    if constexpr (detail::is_any_v<U, view<vector>, view<set>>) {
+      auto equals_lhs = [&](auto y) {
+        if constexpr (std::is_same_v<decltype(lhs), decltype(y)>)
+          return lhs == y;
+        else
+          return false;
+      };
+      auto pred = [&](const auto& rhs_element) {
+        return caf::visit(equals_lhs, rhs_element);
+      };
+      return std::find_if(rhs->begin(), rhs->end(), pred) != rhs->end();
+      return false;
+    } else {
+      // Default case.
+      return false;
+    }
+  }
+
+  bool operator()(view<std::string>& lhs, view<std::string>& rhs) {
+    return rhs.find(lhs) != std::string::npos;
+  }
+
+  bool operator()(view<std::string>& lhs, view<pattern> rhs) {
+    return rhs.search(lhs);
+  }
+
+  bool operator()(view<address> lhs, view<subnet> rhs) {
+    return rhs.contains(lhs);
+  }
+
+  bool operator()(view<subnet> lhs, view<subnet> rhs) {
+    return rhs.contains(lhs);
+  }
+};
```

Comment:
There are some issues with the argument signatures:
1. Sometimes you use a non-const reference, other times a value.
2. Since this is not a mutating function, references should be const
3. I would replace the value with const references or value. If the predicate doesn't get inlined, it might be a good idea to use const references to avoid unnecessary bumping of the ref-count in the type-erased container views.